src/DependencyInjection/LopiPusherExtension.php

```diff
@@ -27,25 +30,26 @@ public function load(array $configs, ContainerBuilder $container)
         $configuration = new Configuration();
         $config = $this->processConfiguration($configuration, $configs);
 
-        $container->setParameter('lopi_pusher.config', $config);
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $loader->load('services.xml');
 
-        if (null !== $config['auth_service_id']) {
-            $container->setAlias('lopi_pusher.authenticator', $config['auth_service_id'])->setPublic(true);
+        if (null === $config['auth_service_id']) {
+            $container->removeDefinition('lopi_pusher.auth_controller');
```

Comment:
If you're doing this, then you don't need the "authenticator" argument to be nullable on the controller and you don't need to check to see if it's null in the controller.